src/parser.rs

```diff
@@ -444,28 +444,24 @@ impl Parser {
         };
         let on_overflow = if self.parse_keywords(vec!["ON", "OVERFLOW"]) {
             let error = self.parse_keyword("ERROR");
-            let filler = if !error {
-                self.expect_keyword("TRUNCATE")?;
-                Some(Box::new(self.parse_expr()?))
+            Some(if error {
+                ListAggOnOverflow::Error
             } else {
-                None
-            };
-            let with_count = if !error {
-                let with_count = self.parse_keywords(vec!["WITH", "COUNT"]);
-                let without_count = self.parse_keywords(vec!["WITHOUT", "COUNT"]);
-                if !with_count && !without_count {
-                    return parser_err!(
-                        "Expected either WITH COUNT or WITHOUT COUNT in LISTAGG".to_string()
-                    );
+                self.expect_keyword("TRUNCATE")?;
+                let filler = Some(Box::new(self.parse_expr()?));
```

Comment:
As implemented, the `filler` isn't optional. I'm fine with this if we change the spec and the AST, or you could peek_token() to see if the next token is either a `WITH` or a `WITHOUT` keyword before attempting to `parse_expr`